collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesActivity.java

```diff
@@ -101,4 +104,17 @@ public void preventOtherWaysOfEditingForm() {
     public void onDialogClosed() {
         startActivityAndCloseAllOthers(this, MainMenuActivity.class);
     }
+
+    @Override
+    public void onPasswordChanged(String password) {
+        SharedPreferences.Editor editor = this.getSharedPreferences(ADMIN_PREFERENCES, MODE_PRIVATE).edit();
```

Comment:
Is there any reason this logic couldn't happen in the fragment? Then we can avoid having to deal with having a listener. We can also then test this functionality in `ChngeAdminPasswordDialogFragmentTest`.

Reply:
Yes I think it would be nice to remove the unnecessary listener. I will revert this change.